src/neurostatslib/basis.py

```diff
@@ -135,51 +113,31 @@ def evaluate_on_grid(self, *n_samples: int) -> Tuple[NDArray, ...]:
 
         Returns
         -------
-            Xs[1], ..., Xs[n] : (n_samples[0], ... , n_samples[n])
-            A tuple containing the meshgrid values, one element for each of the n dimension of the grid, where n equals
-            to the number of inputs.
-            Y : (number of basis, n_samples[0], ... , n_samples[n])
-            the basis function evaluated at the samples.
+        *Xs :
+            A tuple of arrays containing the meshgrid values, one element for each of the n dimension of the grid,
+            where n equals to the number of inputs.
+            The size of Xs[i] is (n_samples[0], ... , n_samples[n]).
+        Y :
+            The basis function evaluated at the samples,
+            shape (number of basis, n_samples[0], ... , n_samples[n]).
 
         """
         self._check_input_number(n_samples)
-        n_basis: int = np.prod(n_samples, dtype=int)
-        self._check_full_model_matrix_size(n_basis)
 
         # get the samples
         sample_tuple = self._get_samples(*n_samples)
         Xs = np.meshgrid(*sample_tuple, indexing="ij")
 
         # call evaluate to evaluate the basis on a flat NDArray and reshape to match meshgrid output
         Y = self.evaluate(*tuple(grid_axis.flatten() for grid_axis in Xs)).reshape(
-            (self._n_basis_funcs,) + n_samples
+            (self._n_basis_funcs, *n_samples)
         )
 
         return *Xs, Y
 
-    def _check_enough_samples(self, eval_basis: NDArray) -> None:
-        """
-        Checks if there are enough samples for evaluation.
-
-        Parameters
-        ----------
-        eval_basis : (number of basis, number of samples)
-            The basis evaluated at the samples
-
-        Raises
-        ------
-        UserWarning
-            If the number of basis sets exceeds the number of samples.
-        """
-        if eval_basis.shape[0] > eval_basis.shape[1]:
-            warnings.warn(
-                "Basis set number exceeds the number of samples! "
-                "Consider reducing the number of basis or increase sample size."
-            )
-
     def _check_input_number(self, xi: Tuple) -> None:
         """
-        Check that the number of inputs provided by the user matches the number of inputs that the Basis object requires.
+        Check that the number of inputs provided by the user matches the number of inputs required.
```

Comment:
I think we talked about possibly changing this function name? to make it clearer that it's about the dimensionality of the basis and input. what did we decide?